src/h5web/App.test.tsx

```diff
@@ -1,23 +1,134 @@
 import React from 'react';
-import { render } from '@testing-library/react';
+import {
+  render,
+  RenderResult,
+  fireEvent,
+  waitFor,
+} from '@testing-library/react';
 import App from './App';
 import MockProvider from './providers/mock/MockProvider';
 
 const DOMAIN = 'bsa_002_000-integrate-sub';
 
-test('renders with root group selected', async () => {
-  const { findByText, findByRole } = render(
+function renderApp(): RenderResult {
+  return render(
     <MockProvider domain={DOMAIN}>
       <App />
     </MockProvider>
   );
+}
+
+test('renders with root group selected', async () => {
+  const { findByText, getByRole } = renderApp();
 
   const noVisText = await findByText('Nothing to visualize');
-  expect(noVisText).toBeInTheDocument();
+  expect(noVisText).toBeVisible();
 
-  const title = await findByRole('heading', { name: DOMAIN });
-  expect(title).toBeInTheDocument();
+  const title = getByRole('heading', { name: DOMAIN });
```

Comment:
If the element you're looking for has a specific semantic, then it's better to look for it with `getByRole` (or `findByRole` or `queryByRole`).